plugins/API/Filter/DataComparisonFilter.php

```diff
@@ -0,0 +1,520 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\API\Filter;
+
+use Piwik\API\Request;
+use Piwik\Common;
+use Piwik\Config;
+use Piwik\DataTable;
+use Piwik\DataTable\Simple;
+use Piwik\Metrics;
+use Piwik\Metrics\Formatter;
+use Piwik\Period;
+use Piwik\Piwik;
+use Piwik\Plugin\Report;
+use Piwik\Plugins\AbTesting\DataTable\Filter\BaseFilter;
+use Piwik\Segment;
+use Piwik\Segment\SegmentExpression;
+
+// TODO: unit test
+
+// TODO: if comparing days w/ non-days, in html table & elsewhere, we must display nb_visits instead of nb_uniq_visitors
+
+/**
+ * TODO
+ */
+class DataComparisonFilter extends BaseFilter
```

Comment:
There are a lot of private methods in the function... maybe some of them could be extracted to some other classes so they can be tested?